src/overmap_location.cpp

```diff
@@ -25,14 +28,48 @@ const overmap_location &string_id<overmap_location>::obj() const
     return locations.obj( *this );
 }
 
+template<>
+inline bool int_id<overmap_location>::is_valid() const
+{
+    return locations.is_valid( *this );
+}
+
+template<>
+const overmap_location &int_id<overmap_location>::obj() const
+{
+    return locations.obj( *this );
+}
+
+template<>
+const string_id<overmap_location> &int_id<overmap_location>::id() const
+{
+    return locations.convert( *this );
+}
+
+template<>
+int_id<overmap_location>::int_id( const string_id<overmap_location> &id ) : _id( id.id() )
+{
+}
+
+template<>
+int_id<overmap_location> string_id<overmap_location>::id() const
```

Comment:
Like in "trap.cpp", the definition of `string_id<>::id` must be before that of `int_id<>::int_id` (as the later uses the former). Generally: follow the order of the functions in "trap.cpp", that should work.

Reply:
Thanks. I have tried that - it is still not working.

It compiles fine in VS and causes runtime exception.

And Travis now has following error `src/overmap_location.cpp:49:73: error: specialization of ‘int_id<T>::int_id(const string_id<T>&) [with T = overmap_location]’ after instantiation
 int_id<overmap_location>::int_id( const string_id<overmap_location> &id ) : _id( id.id() )`.

Reply:
Some more changes and Travis error changes to:

```
19:59:07 src/overmap_location.cpp:47:27: error: explicit specialization of 'int_id' after instantiation
19:59:07 int_id<overmap_location>::int_id( const string_id<overmap_location> &id ) : _id( id.id() )
19:59:07                           ^
19:59:07 src/overmap_location.cpp:42:38: note: implicit instantiation first required here
19:59:07     return locations.convert( *this, string_id<overmap_location>::NULL_ID() );
````